### PR TITLE
Mutant Hat Offset Fix [FIXED FOR REAL THIS TIME]

### PR DESCRIPTION
**Pull request by CliffracerX**
## About The Pull Request
Fixes Teshari being unable to wear hats correctly, introduced in #21001.  Before the introduction of the worn_offset datum with the golem rework, sprite offsets for clothing were being applied with a mutant_override check - which was removed with the offset_features application to make way for worn_offset.  This brings back the check and ensures that bespoke sprites aren't being unnecessarily offset.

Affects ``update_worn_glasses``, ``update_inv_ears``, ``update_worn_neck``, ``update_worn_head``, ``update_worn_belt``, ``update_worn_oversuit``, ``update_worn_mask``, and ``update_worn_back`` in  ``code/modules/mob/living/carbon/human/human_update_icons.dm``.

## How This Contributes To The Skyrat Roleplay Experience
Stops Teshari from having their hats phase into their face & their masks into their chest, among other likely as-of-yet undocumented strangeness.

## Proof of Testing
<details>
<summary>Screenshots/Videos</summary>

image
image
</details>

## Changelog
:cl:
fix: teshari remembered how to wear their own hats again
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
Now check how they are wearing MODs.

**Comment by CliffracerX** [comment]
> Now check how they are wearing MODs.

image

Welp.  This is gonna suck to fix.

**Comment by ghost** [comment]
You can "draw" youre PR by pressing button near the stroke "Still in progress?"

**Comment by ghost** [comment]
Also you can check my commits https://github.com/Skyrat-SS13/Skyrat-tg/pull/21039 . I have tryed to devide checks of the offsets. But it also sucks.

**Comment by CliffracerX** [comment]
image
image
image
image
image

Hats, masks, MOD, all APPEAR to be acting like they did pre-Golem update.

File: code/modules/mob/living/carbon/human/human_update_icons.dm [hunk 1]
```diff
@@ -288,7 +291,10 @@ There are several things that need to be remembered:
 
 		var/mutable_appearance/ears_overlay = ears.build_worn_icon(default_layer = EARS_LAYER, default_icon_file = icon_file, override_file = mutant_override ? icon_file : null) // SKYRAT EDIT CHANGE
 
-		my_head.worn_ears_offset?.apply_offset(ears_overlay)
+		// SKYRAT EDIT ADDITION
+		if(mutant_override == FALSE)
```
Comment (Gandalf2k15) [inline code/modules/mob/living/carbon/human/human_update_icons.dm]:
```suggestion
		if(!mutant_override)
```

and ditto for the rest

Comment (CliffracerX) [reply code/modules/mob/living/carbon/human/human_update_icons.dm]: Done and done.

**Review by lessthnthree — APPROVED** [review]

**Pull request merged**